src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java

```diff
@@ -190,6 +193,9 @@ else if (KerberosGrantConstants.KERBEROS_IDP_SPNPASSWORD.equals(property.getName
                         kerberosUsersId.length();
                 AuthenticatedUser kerberosUser = new AuthenticatedUser();
                 kerberosUser.setUserName(kerberosUsersId.substring(0, indexOfAt));
+                if(StringUtils.isNotEmpty(userstoreDomains)){
```

Comment:
unless we iterate the user stores and make sure that this use is indeed present in the defined userstore domain can we set this as user store domain?

Check https://github.com/wso2-extensions/identity-local-auth-iwa-kerberos/blob/master/components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java#L241 we do this check.